src/Service/CarService.php

```diff
@@ -128,7 +128,7 @@ public function createNewCar(Car $car): Response
             return new Response('Invalid input data: ' . (string) $errors, Response::HTTP_BAD_REQUEST);
         }
 
-        $car->setCreatedAt(new \DateTimeImmutable());
+        $car->setCreatedAt(new \DateTime());
```

Comment:
why ?

Reply:
i changed from using DateTimeImmutable to DateTime because the conversion was not working properly with DateTimeImmutable. when i used DateTimeImmutable, the setCreatedAt field was being shown as null, some problem with converts. i discovered this when I used dd() for debugging. once i switched to DateTime, everything started working correctly, as DateTime is much easier to work with  and the conversion works as expected.